Archives/ClmFile.cpp

```diff
@@ -366,7 +366,7 @@ namespace Archives
 			
 			// If not the right header, skip to next header
 			currentPosition += header.length + 8;
-			fileStreamReader.Seek(currentPosition);
+			seekableStreamReader.Seek(currentPosition);
 		} while (currentPosition < fileSize);
```

Comment:
I wouldn't mind seeing this loop a little less complex, but it is working and I'm happy with the current form. 

On a similar note. Are the structs DataChunk in ClmFile.h and ChunkHeader in ClmFile.cpp representing the same thing? If so, I'd like to combine them.

Thanks,
Brett

Reply:
Hmm, good catch, they are the same. The ChunkHeader name is probably the most accurate and generic, since there is both a format chunk, and a data chunk, at minimum, in a wave file.

As the structs are both declared within functions, one of them will need to move somewhere more globally accessible.

It may make some sense to separate out some of the WAVE file code into a separate class for reading/writing .wav files. I had tried doing that at one point, but found the Stream code wasn't quite advanced enough yet to support it well. In particular, being about to return a slice of a stream would have been useful for providing access to the raw wave data, without the file headers. Ideally, such a stream slice should be bounds checked, so as to prevent seeking or reading outside of the slice bounds.

For now perhaps we can just move the struct up to the top of the .cpp file, or maybe the private section of the header file. Or at the very least, use the same name for the duplicate.

Reply:
I'll use the definition of ChunkHeader from ClmFile::FindChunk by pulling it into the Header file so it can be used in both places.

I would agree on basic read/write of WAVE in a separate class. Not sure if we will make it that far. For now it would be nice to maybe just organize the structs involved in reading a generic wave file separate from the ClmFile struct to make things more clear.